Description: Build against system libraries instead of vendored ones
 This patch changes the QuickBuild configuration output so that the linker
 invocation commands produce binaries which are linked to the system versions
 of liblua5.3 and SPIRV-Cross rather than the vendored project snapshots present
 in the release tarball and typically compiled at build time. This produces a
 RetroArch binary roughly 1 MiB smaller, on average.
Author: Peter J. Mello <admin@petermello.net>
Origin: other, https://git.launchpad.net/~libretro/libretro/+git/RetroArch-debian/commit/patches/0000-use-system-libs.patch?id=00fcf2a6c54be09d8b84443651063c7527f420db
Bug-Debian: https://bugs.debian.org/949708
Bug-Debian: https://bugs.debian.org/1053190
Forwarded: no
Last-Update: 2026-10-14

--- a/Makefile.common
+++ b/Makefile.common
@@ -398,6 +398,7 @@ OBJ += \
 
 ifeq ($(HAVE_VIDEO_FILTER), 1)
 DEFINES += -DHAVE_VIDEO_FILTER
+INCLUDE_DIRS += -I./gfx
 OBJ     += gfx/video_filter.o
 endif
 
@@ -2720,42 +2721,9 @@ ifeq ($(HAVE_NETWORKING), 1)
       endif
 
       ifeq ($(HAVE_LUA), 1)
-         DEFINES += -DHAVE_LUA \
-                    -DLUA_32BITS
-         INCLUDE_DIRS += -Ideps/lua/src
-         OBJ += deps/lua/src/lapi.o \
-                deps/lua/src/lcode.o \
-                deps/lua/src/lctype.o \
-                deps/lua/src/ldebug.o \
-                deps/lua/src/ldo.o \
-                deps/lua/src/ldump.o \
-                deps/lua/src/lfunc.o \
-                deps/lua/src/lgc.o \
-                deps/lua/src/llex.o \
-                deps/lua/src/lmem.o \
-                deps/lua/src/lobject.o \
-                deps/lua/src/lopcodes.o \
-                deps/lua/src/lparser.o \
-                deps/lua/src/lstate.o \
-                deps/lua/src/lstring.o \
-                deps/lua/src/ltable.o \
-                deps/lua/src/ltm.o \
-                deps/lua/src/lundump.o \
-                deps/lua/src/lvm.o \
-                deps/lua/src/lzio.o \
-                deps/lua/src/lauxlib.o \
-                deps/lua/src/lbaselib.o \
-                deps/lua/src/lbitlib.o \
-                deps/lua/src/lcorolib.o \
-                deps/lua/src/ldblib.o \
-                deps/lua/src/liolib.o \
-                deps/lua/src/lmathlib.o \
-                deps/lua/src/loslib.o \
-                deps/lua/src/lstrlib.o \
-                deps/lua/src/ltablib.o \
-                deps/lua/src/lutf8lib.o \
-                deps/lua/src/loadlib.o \
-                deps/lua/src/linit.o
+         DEFINES      += -DHAVE_LUA -DLUA_32BITS -DLUA_USE_LINUX -DLUA_USE_C89
+         INCLUDE_DIRS += -I/usr/include/lua5.3
+         LIBS         += -llua5.3 -lreadline
       else
          DEFINES += -DRC_DISABLE_LUA
       endif
